frontends/common/constantFolding.h

```diff
@@ -162,6 +162,21 @@ class ConstantFolding : public PassManager {
         if (typeMap != nullptr) passes.push_back(new ClearTypeMap(typeMap));
         setName("ConstantFolding");
     }
+
+    /// Function definition for hook to filter which Expression objects are processed.
+    ///
+    /// Function should return nullptr to process the Expression, or a non-null pointer
+    /// to bypass processing. Returned value will be returned by the postorder function.
```

Comment:
Can you give an example how this would be used? Are you trying to modify a behavior in some of the standard pass managers? Or are you modifying it in your midend/backend? For the second case, it would make significantly more sense to derive a backend-specific version of constant folding from the generic one.